omas/omas_physics.py

```diff
@@ -180,6 +181,102 @@ def summary_greenwald(ods, update=True):
+
+    """
+     Calculates line-aveage electron density for each time slice and stores them in the summary ods.
+
+    :param ods: input ods
+    
+    :param line_grid: number of points to calculate line average density over (includes point outside of boundary)
+    
+    :param time_index: time slices to process
+
+    :param update: operate in place
+
+    :return: updated ods
+    """
+    import scipy
+    
+    ods_n = ods
+    if not update:
+        from omas import ODS
+
+        ods_n = ODS().copy_attrs_from(ods)
+
+    if time_index is None:
+        for time_index in range(len(ods['core_profiles']['profiles_1d'])):
+            summary_lineaverage_density(ods_n, line_grid=line_grid, time_index=time_index, update=False)
+        return ods_n
+
+    Rb = ods['equilibrium']['time_slice'][time_index]['boundary']['outline']['r']
+    Zb = ods['equilibrium']['time_slice'][time_index]['boundary']['outline']['z']
+
+
+    Rgrid = ods['equilibrium']['time_slice'][time_index]['profiles_2d'][0]['grid']['dim1']
+    Zgrid = ods['equilibrium']['time_slice'][time_index]['profiles_2d'][0]['grid']['dim2']
+
+    psi2d = ods['equilibrium']['time_slice'][time_index]['profiles_2d'][0]['psi']
+    psi_interp = scipy.interpolate.interp2d(Zgrid, Rgrid, psi2d)
+    psi_eq = ods['equilibrium']['time_slice'][time_index]['profiles_1d']['psi']
+    rhon_eq = ods['equilibrium']['time_slice'][time_index]['profiles_1d']['rho_tor_norm']
+    rhon_cp = ods['core_profiles']['profiles_1d'][time_index]['grid']['rho_tor_norm']
+    ne = ods['core_profiles']['profiles_1d'][time_index]['electrons']['density_thermal']
+    ne = numpy.interp(rhon_eq, rhon_cp, ne)
+    tck = scipy.interpolate.splrep(psi_eq, ne, k=3)
+    
+    if 'time' not in ods['interferometer']:
+        ods_n['interferometer.ids_properties.homogeneous_time'] = 1
+        ods_n['interferometer']['time'] = copy.copy(ods['core_profiles']['time'])
+    
+    ifpaths = [['first_point', 'second_point'], ['second_point', 'third_point',]]
+    for channel in ods['interferometer']['channel']:
+        ne_line_paths = []
+        dist_paths = [] 
+        for ifpath in ifpaths:
+            R1 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[0]]['r']
+            Z1 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[0]]['z']
+            phi1 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[0]]['phi']
+            x1 = R1*numpy.cos(phi1)
+            y1 = R1*numpy.sin(phi1)
+            		
+            R2 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[1]]['r']
+            Z2 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[1]]['z']
+            phi2 = ods['interferometer']['channel'][channel]['line_of_sight'][ifpath[1]]['phi']
+            x2 = R2*numpy.cos(phi2)
+            y2 = R2*numpy.sin(phi2)
+                
+            xline = (x2-x1) * numpy.linspace(0, 1, line_grid) + x1
```

Comment:
```suggestion
            xline = numpy.linspace(x1, x2, line_grid)
```

Reply:
Thats for correcting this. I missed David's comment.